### ISSUE TITLE
🧩 [Phase 3] Pretrained Visual Backbone — Swin-Tiny

### ISSUE THREAD

**Issue by shoaib-inamdar** [shoaib-inamdar/MultiModal-Cane-Disease-Detection#5]
**Parent:** #2 

## 🎯 Goal
Load **Swin-Transformer Tiny** from `timm` as a pretrained feature extractor.
Do NOT build any Vision Transformer from scratch.

## ⚙️ timm Config
```python
import timm
model = timm.create_model(
    "swin_tiny_patch4_window7_224",
    pretrained=True,
    num_classes=0,
    global_pool=""       # ← returns patch tokens, not a pooled vector
)
# Output shape: [B, 49, 768]   (49 = 7×7 spatial patches, 768-dim each)
```

## 📋 Files to Create
- [ ] `src/models/backbone/swin_backbone.py` — `SwinBackbone(nn.Module)`
  - `forward(x)` returns `[B, 49, 768]`
- [ ] `tests/unit/test_swin_backbone.py`
  - `test_output_shape` — assert `output.shape == [B, 49, 768]`
  - `test_gradient_flows` — `loss.backward()` completes without error

## ⚠️ Key Rules
- ✅ Use `pretrained=True` always
- ✅ Use `global_pool=""` to get patch tokens (not global vector)
- ❌ Do NOT build Swin from scratch
- ❌ Do NOT set `num_classes > 0` (that adds a classification head we don't want)

## ✅ Done When
```python
backbone = SwinBackbone()
out = backbone(torch.rand(2, 3, 224, 224))
assert out.shape == torch.Size([2, 49, 768])
```

**Comment by Flash-45** [comment]
@shoaib-inamdar could you please assign me this issue
